runtime/common/addresslocation.go

```diff
@@ -85,6 +85,10 @@ func (l AddressLocation) Description() string {
 	)
 }
 
+func (l AddressLocation) ID() string {
+	return fmt.Sprintf("%v.%v", AddressLocationPrefix, l.String())
```

Comment:
Can use string formatting here and for the other cases:

```suggestion
	return fmt.Sprintf("%s.%s", AddressLocationPrefix, l)
```